scene/gui/spin_box.cpp

```diff
@@ -288,12 +302,18 @@ void SpinBox::_line_edit_editing_toggled(bool p_toggled_on) {
 			line_edit->select_all();
 		}
 	} else {
+		accepted = true;
+
 		// Discontinue because the focus_exit was caused by canceling or the text is empty.
 		if (Input::get_singleton()->is_action_pressed("ui_cancel") || line_edit->get_text().is_empty()) {
 			_update_text();
 			return;
 		}
 
+		if (line_edit->get_text().ends_with(".")) {
+			line_edit->set_text(line_edit->get_text().trim_suffix("."));
```

Comment:
Text input ending in a decimal and submitted that way will assume just the whole number, even if update_on_text_changed is false.